pkgs/by-name/vm/vmware-workstation/package.nix

```
@@ -99,11 +98,11 @@ stdenv.mkDerivation rec {
     ++ lib.optionals enableInstaller [ bzip2 sqlite readline70_compat63 ]
     ++ lib.optionals enableMacOSGuests [ unzip ];
 
-  src = fetchzip {
-    url = "${baseUrl}/core/VMware-Workstation-${version}-${build}.x86_64.bundle.tar";
-    sha256 = "sha256-VzfiIawBDz0f1w3eynivW41Pn4SqvYf/8o9q14hln4s=";
-    stripRoot = false;
-  } + "/VMware-Workstation-${version}-${build}.x86_64.bundle";
+  src = requireFile {
+    name = "VMware-Workstation-Full-${version}-${build}.x86_64.bundle";
```

With your solution, users will need to extract the downloaded *.bundle.tar file and add the *.bundle file to the nix store. What about using the unextracted file (later a `tar` command needs to be added to the unpack phase if you'd like to go with this)?
Also does the new version has `Full` in the filename? I've only tested 17.6.1 where this is not the case.

```suggestion
    name = "VMware-Workstation-${version}-${build}.x86_64.bundle.tar";
```

---

Oh that's neat, thanks for clarifying!